tornado/ioloop.py

```diff
@@ -440,7 +440,9 @@ def _setup_logging(self) -> None:
                 logging.getLogger("tornado.application").handlers,
             ]
         ):
-            logging.basicConfig()
+            # same as logging.basicConfig() but doesn't affect user's logger. issue #2961
+            logger = logging.getLogger("tornado")
+            logger.addHandler(logging.StreamHandler())
```

Comment:
I don't understand what this is for. Why is it needed? I would expect we could just remove the entire `_setup_logging` method and rely on the default behavior (which propagates messages from the application logger to the root logger and then to the last-resort logger).